### PR TITLE
correct the java version that jenkins depends on

### PR DESCRIPTION
**Pull request by ryenus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Jenkins does not yet support Java 9, see https://issues.jenkins-ci.org/browse/JENKINS-40689. 
Therefore we need to explicitly use java 8 at the moment.

* use `/usr/libexec/java_home` to detect java 1.8 and use it

### CODE REVIEW

**Comment by ryenus** [comment]
See: 
* https://issues.jenkins-ci.org/browse/JENKINS-40689
* https://jenkins.io/blog/2016/03/30/usage-statistics-privacy-advisory/

**Comment by ryenus** [comment]
rebased onto latest jenkins 2.94 (commit https://github.com/Homebrew/homebrew-core/commit/5551b9172aa5e5e5e9abe6088523e6ee64e51f11), possible to merge? @ilovezfs

File: Formula/jenkins.rb [hunk 1]
```diff
@@ -40,8 +40,13 @@ def plist; <<~EOS
...
         <key>ProgramArguments</key>
         <array>
-          <string>/usr/bin/java</string>
+          <string>/usr/libexec/java_home</string>
+          <string>-v</string>
+          <string>1.8</string>
+          <string>--exec</string>
+          <string>java</string>
           <string>-Dmail.smtp.starttls.enable=true</string>
+          <string>-Dhudson.model.UsageStatistics.disabled=true</string>
```
Comment (fxcoudert) [inline Formula/jenkins.rb]: Why this unrelated change?

Comment (ryenus) [reply Formula/jenkins.rb]: Because it's better for users to explicitly opt-in telemetry: https://wiki.jenkins.io/display/JENKINS/Usage+Statistics

**Comment by ryenus** [comment]
@fxcoudert PR updated, only java version requirement related change now.

**Comment by ilovezfs** [comment]
Thanks @ryenus! Nice work. Mind doing the same for `jenkins-lts`?

**Comment by ryenus** [comment]
Thanks @ilovezfs, sure for jenkins-lts, here it is: https://github.com/Homebrew/homebrew-core/pull/22671

Besides the plist file, I've also patched the launcher script, however this requires another PR: https://github.com/Homebrew/brew/pull/3656
